stistools/add_s_region.py

```diff
@@ -0,0 +1,403 @@
+        xmax = min(wcslimits[1], siaflimits[1])
+    ymin = max(wcslimits[2], siaflimits[2])
+    ymax = min(wcslimits[3], siaflimits[3])
+    return (np.array([xmin, xmin, xmax, xmax, xmin]),
+            np.array([ymin, ymax, ymax, ymin, ymin]))
+
+def coords_from_s_region(s_region):
+    """Helper function to extract the RA and DEC as lists of floating
+    point numbers from the S_REGION string"""
+
+    first_coordinate = 0
+    coords = s_region.split()
+    while coords[first_coordinate] in ['POLYGON', 'ICRS']:
+        first_coordinate = first_coordinate + 1
+    ra = coords[first_coordinate::2]
+    ra = [float(x) for x in ra]
+    dec = coords[first_coordinate+1::2]
+    dec = [float(x) for x in dec]
+    return (ra, dec)
+
+def write_region_file(file, include_mast=True):
+    """Convenience function to write a DS9 regions file from the S_REGION
+    value in the header of a file.  If include_mast is set to True, will
+    also get the existing S_REGION value from the mast catalog and include
+    it in the region file.  The region file will include the keyword S_REGION
+    in blue, the RA_APER and DEC_APER in red, and the mast catalog value of the
+    S_REGION in green, if include_mast is True"""
+
+    f1 = fits.open(file)
+    rootname = f1[0].header['ROOTNAME']
+    ra_aper = f1[1].header['RA_APER']
+    dec_aper = f1[1].header['DEC_APER']
+    s_region = f1[1].header['S_REGION']
+    f1.close()
+    f_region = open(rootname+'.reg', mode='w')
+    text = 'icrs; polygon'
+    ra, dec = coords_from_s_region(s_region)
+    for x, y in zip(ra, dec):
+        text = text + ' {} {}'.format(x, y)
+    text = text + " # color=blue"
+    f_region.write(text)
+    f_region.write('\n')
+    text = 'icrs; point' + ' {} {}'.format(ra_aper, dec_aper) + ' # color=red \n'
+    # Get existing S_REGION value from MAST
+    if include_mast:
+        from astroquery.mast import Observations
+        row = Observations.query_criteria(obs_id=rootname)
+        existing_s_region = row['s_region'][0]
+        f_region.write(text)
+        text = 'icrs; polygon'
+        ra, dec = coords_from_s_region(existing_s_region)
+        for x, y in zip(ra, dec):
+            text = text + ' {} {}'.format(x, y)
+        text = text + " # color=green"
+        f_region.write(text)
+        f_region.write('\n')
+    f_region.close()
+    return
+
+
+
+def main(dry_run=False):
+    files_to_process = get_files_to_process()
+    for file in files_to_process:
+        add_s_region(file, dry_run=dry_run)
+    return
+
+if __name__ == '__main__':
+
+    parser = argparse.ArgumentParser(
+    """Add S_REGION value to raw data headers"""
+    )
+
+    parser.add_argument(
+        '--dry_run', action='store_true',
+        help="Calculate S_REGION value, but don't write to data header[s]")
+
+    parser.add_argument('-v', '--version', help='Print version info', action='version',
+    version = __version__)
+
+    args = parser.parse_args()
+
+    if '--version' in args:
```

Comment:
It looks like argparse exits at `parser.parse_args()` without the need to explicitly call `sys.exit(0)`.

(Also note that the key would be stored as `version`, not `--version`, but it never reaches this test.)

Reply:
I wasn't aware of that behaviour of argparse - good to know!